src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -292,6 +301,26 @@ internal static SecurityKey GetSecurityKey(EncryptingCredentials encryptingCrede
 
                 securityKey = encryptingCredentials.Key;
             }
+#if NET472
+            else if (SupportedAlgorithms.EcdsaWrapAlgorithms.Contains(encryptingCredentials.Alg))
+            {
+                // on decryption we get the public key from the EPK value see: https://datatracker.ietf.org/doc/html/rfc7518#appendix-C
+                string apu = null, apv = null;
+                if (additionalHeaderClaims != null && additionalHeaderClaims.Count > 0)
+                {
+                    if (additionalHeaderClaims.TryGetValue(JwtHeaderParameterNames.Apu, out object objApu))
+                        apu = objApu.ToString();
+                    if (additionalHeaderClaims.TryGetValue(JwtHeaderParameterNames.Apv, out object objApv))
+                        apv = objApv.ToString();
+                }
```

Comment:
objApu and objApv can be null.

Reply:
Added clause for null and a unit test case for that scenario in [1595e40](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1834/commits/1595e4097fc1b5c0f5b7a00fccf7f878bdd9c202)